scout/server/blueprints/variants/views.py

```diff
@@ -282,170 +281,58 @@ def str_variants(institute_id, case_name):
 @templated("variants/sv-variants.html")
-def sv_variants(institute_id, case_name):
-    """Display a list of structural variants."""
-    page = controllers.get_variants_page(request.form)
-    category = "sv"
-    institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
-    variant_type = Markup.escape(
-        request.args.get("variant_type", request.form.get("variant_type", "clinical"))
-    )
-    if variant_type not in ["clinical", "research"]:
-        variant_type = "clinical"
-    variants_stats = store.case_variants_count(case_obj["_id"], institute_id, variant_type, False)
-
-    controllers.set_hpo_clinical_filter(case_obj, request.form)
+def sv_variants(institute_id: str, case_name: str):
+    """Display a list of structural variants (SV)."""
 
-    if "dismiss_submit" in request.form:  # dismiss a list of variants
-        controllers.dismiss_variant_list(
-            store,
-            institute_obj,
-            case_obj,
-            "variant.sv_variant",
-            request.form.getlist("dismiss"),
-            request.form.getlist("dismiss_choices"),
+    def form_builder(store, inst, case, cat, vtype):
+        """Builds the SV filters form."""
+        return controllers.populate_sv_mei_filters_form(
+            store=store, institute_obj=inst, case_obj=case, category=cat, request_obj=request
         )
 
-    # update status of case if visited for the first time
-    controllers.activate_case(store, institute_obj, case_obj, current_user)
-    form = controllers.populate_sv_filters_form(store, institute_obj, case_obj, category, request)
-
-    # Populate chromosome select choices
-    controllers.populate_chrom_choices(form, case_obj)
-
-    # Populate custom soft filters
-    controllers.populate_institute_soft_filters(form=form, institute_obj=institute_obj)
-
-    genome_build = get_case_genome_build(case_obj)
-    cytobands = store.cytoband_by_chrom(genome_build)
-
-    controllers.update_form_hgnc_symbols(store, case_obj, form)
-
-    variants_query = store.variants(
-        case_obj["_id"], category=category, query=form.data, build=genome_build
+    return controllers.render_variants_page(
+        category="sv",
+        institute_id=institute_id,
+        case_name=case_name,
+        form_builder=form_builder,
     )
 
-    result_size = store.count_variants(
-        case_obj["_id"], form.data, None, category, build=genome_build
-    )
 
-    # if variants should be exported
-    if request.form.get("export"):
-        return controllers.download_variants(store, case_obj, variants_query)
+@variants_bp.route("/<institute_id>/<case_name>/cancer/sv-variants", methods=["GET", "POST"])
+@templated("variants/cancer-sv-variants.html")
+def cancer_sv_variants(institute_id: str, case_name: str):
+    """Display a list of cancer structural variants."""
 
-    data = controllers.sv_mei_variants(store, institute_obj, case_obj, variants_query, page)
+    def form_builder(store, inst, case, cat, vtype):
```

Comment:
I was thinking also about this line, to get typing, but this is already better with thes controller helper call safer.